test/powershell/engine/Basic/Assembly.LoadNative.Tests.ps1

```diff
@@ -0,0 +1,117 @@
+
+    BeforeAll {
+
+        ## The assembly files cannot be removed once they are loaded, unless the current PowerShell session exits.
+        ## If we use $TestDrive here, then Pester will try to remove them afterward and result in errors.
+        $TempPath = [System.IO.Path]::GetTempFileName()
+        if (Test-Path $TempPath) { Remove-Item -Path $TempPath -Force -Recurse }
+        New-Item -Path $TempPath -ItemType Directory -Force > $null
+
+        $root = Join-Path $TempPath "testDllNativeFolder"
+        New-Item -Path $root -ItemType Directory -Force > $null
+
+        $dllSourceFolder = Join-Path $PSScriptRoot "assets"
+        Copy-Item -Recurse -Path $dllSourceFolder\* -Destination $root
+
+        $source = @"
+            using System;
+            using System.Runtime.InteropServices;
+            public class TestNativeClass
+            {
+                public static int Add(int a, int b)
+                {
+                    return (a + b);
+                }
+
+                public static IntPtr LoadNative()
+                {
+                    return sk_codec_min_buffered_bytes_needed();
+                }
+
+                // Copied from https://github.com/mono/SkiaSharp/blob/e0f57880ca5eadfaddde520e8d8365bc00b91d5d/binding/Binding/SkiaApi.generated.cs
+                [DllImport ("libSkiaSharp", CallingConvention = CallingConvention.Cdecl)]
+                internal static extern /* size_t */ IntPtr sk_codec_min_buffered_bytes_needed ();
+            }
+"@
+
+        Add-Type -OutputAssembly $root\managed.dll -TypeDefinition $source
+
+        Add-Type -Assembly $root\managed.dll
+    }
+
+    It "Can load native dll" {
+        # Managed dll is loaded
+        [TestNativeClass]::Add(1,2) | Should -Be 3
+
+        # Native dll is loaded from the same managed dll
+        [TestNativeClass]::LoadNative() | Should -Not -BeNullOrEmpty
+    }
+}
+
+Describe "Can load a native assembly" -Tags "CI" {
+
+    BeforeAll {
+
+        ## The assembly files cannot be removed once they are loaded, unless the current PowerShell session exits.
+        ## If we use $TestDrive here, then Pester will try to remove them afterward and result in errors.
+        $TempPath = [System.IO.Path]::GetTempFileName()
+        if (Test-Path $TempPath) { Remove-Item -Path $TempPath -Force -Recurse }
+        New-Item -Path $TempPath -ItemType Directory -Force > $null
+
+        $root = Join-Path $TempPath "testDllNativeFolder"
+        New-Item -Path $root -ItemType Directory -Force > $null
+
+        if ($IsWindows) {
+            $arch = "win-x64"
+            $nativeDllName = "nativedll.dll"
+        } elseif ($IsLinux) {
+            $arch = "linux-x64"
+            $nativeDllName = "nativedll.so"
+        } elseif ($IsMacOs) {
+            $arch = "osx-x64"
+            $nativeDllName = "nativedll.dylib"
+        } else {
+            throw "Unsupported OS architecture"
```

Comment:
We do run tests on arm, arm64 and x86, though not as frequently as CIs. This test should not throw when running on other platforms.

Reply:
Fixed.